Updated Meter Dispose guidance (#37170)
dotnet/docs: docs/core/diagnostics/metrics-instrumentation.md

});
```

> ### Best Practices

Reviewer: ```suggestion
### Best practices
```

> - The <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> does implement IDisposable, but IMeterFactory will manage the lifetime of any Meters it

Reviewer 2: - <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meters it
  creates, disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and will not have any effect.

> - <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meters it

Reviewer: ```suggestion
- <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any `Meter` objects it
```

>   creates, disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and won't have any effect.

Reviewer: ```suggestion
  creates, disposing them when the DI container is disposed. It's unnecessary to add extra code to invoke `Dispose()` on the `Meter`, and it won't have any effect.
```

> creates, automatically disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and won't have any effect.

Author: Yes, but that probably belongs in API docs for IMeterFactory rather here. Could you add it?

> - The <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> does implement IDisposable, but IMeterFactory will manage the lifetime of any Meters it
> creates, automatically disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and won't have any effect.

Reviewer 2: ```suggestion
- <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meters it
  creates, disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and will not have any effect.
```
